web/modules/custom/asset_distribution/asset_distribution.behat.inc

```diff
@@ -299,4 +299,46 @@ class AssetDistributionSubContext extends DrupalSubContextBase implements Drupal
     return $fields;
   }
 
+  /**
+   * Checks if a distribution in compact view contains download links.
+   *
+   * @param string $distribution
+   *    The name of the distribution.
+   *
+   * @Then I should see the download link in the :distribution( asset) distribution
+   *
+   * @throws \Exception
+   *   If region or link within it cannot be found.
+   */
+  public function assertLinkDistribution($distribution) {
+    $distribution = $this->getAssetDistributionByName($distribution);
+    if (empty($distribution->field_ad_access_url->first())) {
+      throw new \Exception(sprintf('Asset distribution "%s" does not contain any download links.', $distribution->label()));
+    }
+    $promoted_link = $distribution->field_ad_access_url->first()->uri;
+    $result = $this->getSession()->getPage()->find('xpath', "//a[@href='{$promoted_link}']");
```

Comment:
This XPath does not actually check if the download link is in the asset distribution, it could be anywhere on the page.

For this to work we need to show the asset distribution title in the compact view. For the moment the placeholder title is still shown.